core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -256,9 +280,11 @@ class WorkerAutomation(
         val workableTiles = currentTile.getTilesInDistance(4)
                 .filter {
                     (it.civilianUnit == null || it == currentTile)
-                            && tileCanBeImproved(unit, it)
-                            && it.getTilesInDistance(2)
-                            .none { tile -> tile.isCityCenter() && tile.getCity()!!.civInfo.isAtWarWith(civInfo) }
+                            && (tileCanBeImproved(unit, it) || it.isPillaged())
+                            && it.getTilesInDistance(2)  // don't work in range of enemy cities
+                        .none { tile -> tile.isCityCenter() && tile.getCity()!!.civInfo.isAtWarWith(civInfo) }
+                            && it.getTilesInDistance(4)  // don't work in range of enemy units
+                        .none { tile -> tile.militaryUnit != null && unit.civInfo.isAtWarWith(tile.militaryUnit!!.civInfo) && tile.militaryUnit!!.movement.canReach(it) }
```

Comment:
`unit.movement.canReach()` is a very costly function, I wouldn't call this for every AI worker; it feels like that could add up to a lot of time in late-game when there are a lot of workers and units.

Reply:
Removed